src/system.h

```diff
@@ -47,8 +53,36 @@ struct Image {
     void Purge() {
         auto seconds_passed = wxGetLocalTime() - last_display;
         if (seconds_passed > 10) bm_display = wxNullBitmap;
-        // TODO: we could even purge bm_orig for even greater memory savings if we have png_data.
     }
+
+
+    vector<uint8_t> ConvertImageToVector(const wxImage &im, wxBitmapType bmt) {
+        vector<uint8_t> pidv;
+        wxMemoryOutputStream mos;
+        off_t beforeimage = mos.TellO();
+        im.SaveFile(mos, bmt);
+        off_t afterimage = mos.TellO();
+        auto sz = afterimage - beforeimage;
+        wxMemoryInputStream mis(mos);
+        pidv.resize(sz);
+        mis.Read(pidv.data(), pidv.size());
+        return pidv;
+    }
+
+    wxImage ConvertVectorToImage(vector<uint8_t> pidv, wxBitmapType bmt) {

```

Comment:
you're passing a vector by value, its going to get copied.. please pass by ref

Reply:
Thanks for this remark! Solved with 1627303bcda085888f066350bb1a6732174f443d